### PR TITLE
Svelte Components

### PR DESCRIPTION
**Pull request by katelkq**
Made more stuff (recent activity, asset, watchlist items) into components, pretty basic still

### CODE REVIEW

File: frontend/src/routes/portfolio/+page.svelte [hunk 1]
```diff
@@ -1,11 +1,19 @@
...
     <h1 class="title is-2">Portfolio</h1>
 </div>
 
 <div class="columns mt-5" style="width: calc(100vw + 12px);">
     <div class="column is-two-fifths">
         <h2 class="title is-5 ml-5">Summary</h2>
-        <div class="box mx-5 has-background-grey-darker" style="height: 25vh;">
+        <div
+            class="box mx-5 has-background-grey-darker"
+            style="height: 25vh; overflow-y: scroll;"
```
Comment (mileswatson) [inline frontend/src/routes/portfolio/+page.svelte:15]: This leaves ugly black bars everywhere (at least on Firefox) - replacing all `overflow-y: scroll;`with `overflow-y: auto` seems to help. Also, I don't think this box needs to be auto-resizable with height: I think making the entire page scrollable (aside from navbar) would work better than compressing all the elements.

File: frontend/src/components/Activity.svelte [hunk 2]
```diff
@@ -0,0 +1,30 @@
+<script lang="ts">
+    interface Activity {
+        time: string;
+        symbol: string;
+        side: string;
+        qty: number;
+        price: number;
+        totalVal: number;
+        status: string;
```
Comment (mileswatson) [inline frontend/src/components/Activity.svelte:9]: Don't make this a string - instead use `status: "success" | "rejected"`;

File: frontend/src/components/Activity.svelte [hunk 3]
```diff
@@ -0,0 +1,30 @@
...
+    }
+
+    export let data: Activity;
+
+    let color = "warning";
```
Comment (mileswatson) [inline frontend/src/components/Activity.svelte:14]: Not great type checking / exhaustivity. A better way could be: 
```ts
const statusColors = {
    success: "success",
    rejected: "danger",
    pending: "warning",
};

$: color = statusColors[data.status];
```

File: frontend/src/routes/portfolio/+page.svelte [hunk 4]
```diff
@@ -1,11 +1,19 @@
+<script>
```
Comment (mileswatson) [inline frontend/src/routes/portfolio/+page.svelte:1]: There's no type checking on this page - remember to use `lang="ts"` on the script tag.

File: frontend/src/components/Asset.svelte [hunk 5]
```diff
@@ -0,0 +1,72 @@
+<script lang="ts">
+    function round(value: number): number {
```
Comment (mileswatson) [inline frontend/src/components/Asset.svelte:2]: Could instead use `.toFixed(2)` for rounding to two dp?

File: frontend/src/components/Asset.svelte [hunk 6]
```diff
@@ -0,0 +1,72 @@
...
+
+    export let data: Asset;
+
+    const colorToday = data.glToday >= 0 ? "success" : "warning";
+    const colorOverall = data.glOverall >= 0 ? "success" : "warning";
+</script>
+
+<div class="block">
+    <div class="columns mx-2 is-vcentered" style="margin: auto;">
+        <div class="column is-2 has-text-centered">
```
Comment (mileswatson) [inline frontend/src/components/Asset.svelte:23]: What's the purpose of is-2 here?

File: frontend/src/components/Watchlist.svelte [hunk 7]
```diff
@@ -0,0 +1,28 @@
...
+    const color = data.chg >= 0 ? "success" : "warning";
+    let selected = false;
+
+    function click() {
+        selected = !selected;
+    }
+</script>
+
+<tr
+    class={selected ? "is-selected" : ""}
```
Comment (mileswatson) [inline frontend/src/components/Watchlist.svelte:20]: class:is-selected={selected}

File: frontend/src/components/Watchlist.svelte [hunk 8]
```diff
@@ -0,0 +1,28 @@
...
+    style="cursor: pointer;"
+    on:click={() => click()}
```
Comment (mileswatson) [inline frontend/src/components/Watchlist.svelte:22]: I believe you can simply do `on:click={click}`

File: frontend/src/routes/explore/+page.svelte [hunk 9]
```diff
@@ -38,30 +39,38 @@
...
-                        </tr>
-                        <tr style="cursor: pointer;">
-                            <th class="has-text-left">NFLX</th>
-                            <td class="has-text-right">361.54</td>
-                            <td class="has-text-right has-text-warning"
-                                >-1.41</td>
-                            <td class="has-text-right has-text-warning"
-                                >-0.39%</td>
-                        </tr>
+                        <Watchlist
```
Comment (mileswatson) [inline frontend/src/routes/explore/+page.svelte:42]: Maybe define an array in the script portion, and then {#each} here?

File: frontend/src/routes/portfolio/+page.svelte [hunk 10]
```diff
@@ -54,83 +62,39 @@
...
-                            <th class="has-text-left">AAPL</th>
-                            <td class="has-text-left">Buy</td>
-                            <td class="has-text-left">10</td>
-                            <td class="has-text-right">151.66</td>
-                            <td class="has-text-right">1516.63</td>
-                            <td class="has-text-left has-text-danger"
-                                >Rejected</td
-                            >
-                        </tr>
+                        <Activity
```
Comment (mileswatson) [inline frontend/src/routes/portfolio/+page.svelte:65]: Same thing about the {#each} here

File: frontend/src/routes/portfolio/+page.svelte [hunk 11]
```diff
@@ -144,67 +108,54 @@
...
-            ornare nisl sodales venenatis. Vivamus id elementum massa. Etiam
-            posuere tortor vel sem volutpat, non dignissim leo ullamcorper.
-            Curabitur pulvinar magna et feugiat facilisis. Sed maximus leo mi,
-            id ultricies tortor rhoncus ut. Morbi sollicitudin arcu et congue
-            viverra. Proin eget iaculis enim. Vestibulum ante ipsum primis in
-            faucibus orci luctus et ultrices posuere cubilia curae; Proin nec
-            rutrum lacus, ac porta quam. Nullam sed ligula hendrerit, hendrerit
-            massa et, facilisis sapien.
-        </p>
+        <Asset
```
Comment (mileswatson) [inline frontend/src/routes/portfolio/+page.svelte:111]: And likewise {#each} here.

**Review by mileswatson — CHANGES_REQUESTED** [review]
Comments noted above.

File: frontend/src/routes/portfolio/+page.svelte (hunk 1 above)
Comment (katelkq) [reply frontend/src/routes/portfolio/+page.svelte:15]: Would you mind sending a screenshot of what it looks like on your browser to Discord? I can't seem to trigger the black bars on my machine...
I will be replacing the `scroll` with `auto`.
With regards to box auto-resizing: valid point, I'll try to made their height immune to browser resize. I'm less sure about making the entire page scrollable though - maybe talk about it on Tues?

File: frontend/src/routes/portfolio/+page.svelte (hunk 4 above)
Comment (katelkq) [reply frontend/src/routes/portfolio/+page.svelte:1]: Oops my mistake

File: frontend/src/components/Asset.svelte (hunk 6 above)
Comment (katelkq) [reply frontend/src/components/Asset.svelte:23]: Column width 2 out of 12

File: frontend/src/components/Activity.svelte (hunk 3 above)
Comment (katelkq) [reply frontend/src/components/Activity.svelte:14]: Will do & thanks!

File: frontend/src/components/Asset.svelte (hunk 6 above)
Comment (mileswatson) [reply frontend/src/components/Asset.svelte:23]: Huh, learnt something new. I’ve always just used the is-x-yths syntax 😅

File: frontend/src/components/Asset.svelte (hunk 5 above)
Comment (katelkq) [reply frontend/src/components/Asset.svelte:2]: Ohh didn't know that before

File: frontend/src/routes/portfolio/+page.svelte (hunk 1 above)
Comment (mileswatson) [reply frontend/src/routes/portfolio/+page.svelte]: Have sent messages in the discord. Also, good job on the portfolio page - resizing works much better now!

**Comment by mileswatson** [comment]
I think login / signup should be switched (generally standard practice to have sign up be standout and login be less standout, and if that's the case then have signup to the left of login unbalances the UI), and I think that the pink login kinda looks ugly, but these are all minor things that can be changed later, so approved :)

**Pull request merged**